_posts/2024_07_07-EOF.md

```diff
@@ -0,0 +1,101 @@
+---
+layout:        post
+title:         Why I am against EOF in Pectra
+date:          2024-07-12 19:00:00
+summary:       This post explains the advantages and disadvantages of EOF and my thoughts about inclusion in the Prague/Electra hardfork.
+categories:    blog
+tags:          ethereum pectra EOF testing
+---
+
+The Ethereum Object Format (EOF), is an upgrade to the EVM that has been worked on for roughly for years and is included in Pectra. I am not convinced that it is very useful and thus have been arguing against it in the past. I'm writing this post to discuss some of the advantages and disadvantages of EOF and why I think that it shouldn't be included.
+
+During this post, I will refer to the current EVM as "legacy EVM". This is just to make the distinction between the EOF EVM and the current EVM easier. It should not be read as "the current EVM is outdated".  
+
+# Advantages
+
+EOF brings several advantages.
+
+## Stack To Deep
+
+One big annoyance of Solidity programmers has been the stack to deep issue. Since the EVM only supports DUP1-DUP16 and SWAP1-SWAP16, opcodes to swap stack elements around, you can't have more than 16 local variables or function parameters in Solidity. This has been a big annoyance for developers as they need to structure their code around this issue. 
+
+EOF introduces DUPN, SWAPN and EXCHANGE, three opcodes that take an 8-bit immediate which makes it possible to have up to 256 local variables or function parameters. 
+However DUPN, SWAPN and EXCHANGE are not EOF exclusive, so they could very easily be added to the normal EVM without EOF.
+
+## Remove JUMPDEST analysis
+
+Probably the biggest advantage of EOF is that the JUMPDEST analysis is performed at deploy time instead of at runtime. The JUMPDEST analysis checks which parts of the byte code are data and which are valid jump destinations. This needs to be done, so that a JUMP opcode does not end up in the immediate of for example a PUSH32. 
```

Comment:
```suggestion
Probably the biggest advantage of EOF is that the JUMPDEST analysis is performed at deploy time instead of at runtime. The JUMPDEST analysis checks which parts of the bytecode are data and which are valid jump destinations. This needs to be done so that a JUMP opcode does not end up in the immediate of, for example, a PUSH32. 
```